### ISSUE TITLE
The watch option for a whole folder outputs a single file instead of multiple HTML files

### ISSUE THREAD

**Issue by samy** [mjmlio/mjml#1014]
If I specify something else than "current folder" (using dot), the compilation succeed but creates single file containing all the resulting HTML of the source folder MJML files.

**Reproduction Steps:**

1. Use the following command:
`mjml -w  ./whatever/*.mjml -o ./outputfolder/`

**Expected behavior:**

If I understood all the issues and doc, I should find in `outputfolder` an HTML file for each __.mjml__ file of my source folder

**Observed behavior:**

The command creates a `outputfolder` file.
This file content is not updated if I edit one of the MJML files of the source folder **but** MJML detects all changes on the source folder and says "Successfully completed" for each one (without updating output).

**MJML version:**

4.0.0

**Node version:**

8.9.4

**Comment by iRyusa** [comment]
Hi @samy 
To avoid any issue, the cli doesn't create the folder, you have to create it first. Can you try it back like that ?

If you're using ZSH you might need to escape the `*` char too

**Comment by samy** [comment]
If I create the folder before using the command, the folder stays empty and the command output is:
`C:\pathtomyfile\test.mjml - Error while compiling file`

**Comment by bensreed** [comment]
I think I'm experiencing something related to @samy:

When attempting to build mjml using NPM Scripts with multiple input files and includes, with a custom output folder, the build fails without error.

  **Reproduction Steps:**

  1. In my package.json in the `scripts` section: 
`"email-build": "mjml ./assets/templates/mjml/*.mjml -o ./assets/templates/emails/",`
  2. `npm run email-build`

  **Expected behavior:**
All **.mjml** files included in input folder are compiled and written to output folder defined

  **Observed behavior:**
Build fails with the following error:

```
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! email-build: `mjml ./assets/templates/mjml/*.mjml -o ./assets/templates/emails/`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the  email-build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

As far as I can tell, it is failing because `replaceExtension` function in **outputToFile.js** in the CLI pacakge is returning the full path in this case, so I end up with a path like  `./assets/templates/mjml/assets/templates/emails/welcome` if `welcome.mjml` is a given template, causing the file to not write and the build to error out.

FWIW, the client errors out at `case 37` after sending files to `outputToFile` in `client.js` 

If I change the `replaceExtension` function to replace **.mjml** with **.html** and drop the rest of the path, this appears to work, but I haven't tested with includes or files without an output directory specified. 

```javascript
var replaceExtensionForCustomOutputLocation = function replaceExtension(input) {
  var replace = input.replace('.mjml', '.html');
  var output = replace.match(/\w+(?:\.\w+)*$/g);
  return output[0];
};
```

I know this isn't really a solution, but I hope this helps in some way. 

  **MJML version:**

 4.0.0

**Node version:**

v8.1.2

**Comment by iRyusa** [comment]
Hi @bensreed 

Can you open a new issue because it's not really related to the watch issue, we'll take a look at it 👍thanks for the debugging stuff it will help us a lot 💯  

@samy can you try with a minimalist template so we can see if it's an issue with your MJML or not ?

**Comment by kmcb777** [comment]
#1023's fix has been released, and the fix for #1065 (more debug information) will be included in version 4.1.0
It fixes the problems that are described here so i'll close this one
